src/db.js

```diff
@@ -0,0 +1,273 @@
+
+//var depositortest = require('../db/index').depositortest;
+//depositortest.findAll();*
+const { checkDeposit } = require('./api.js');
+const { Pool } = require('pg');
+let pool = new Pool({
+    user: 'hamzaasaad',
+    host: 'localhost',
+    database: 'hamzaasaad',
+    password: null,
+    port: 5431,
+  })
+pool.connect();
+
+pool.query('SELECT NOW()', (err, res) => {
+    if(err){
+        console.log('Database connection failed',err);
+    }
+    else {
+        console.log('Database connected!');
+    }
+  });
+
+const createTable = `create table depositortest(
+    address VARCHAR,
+    norequests INT,
+    dailyCount INT,
+    weeklyCount INT,
+    firstrequesttime TIMESTAMP WITHOUT TIME ZONE,
+    dailyTime TIMESTAMP WITHOUT TIME ZONE,
+    weeklyTime TIMESTAMP WITHOUT TIME 100000000000000ZONE
+
+)`
+
+const depositAmount = "1000000000000000"; //should be 32000000000000000000
+const dailyLimit = 0.002;
+const weeklyLimit = 0.004;
+
+
+
+module.exports = {
+    confirmTransaction: async function(address, topUpAmount){
+        //add try catch
+
+        var addressDetails = (await checkAddressExists(address));
+        //console.log("Check account exists address details:",addressDetails);
+        //Assumes addressDetails will always be an array
+        if (!addressDetails.length){
+            const addressDetails = await setDepositor(address);
+            await updateCounts(addressDetails, topUpAmount);
+            return true
+        }
+        addressDetails = addressDetails[0];
+        //refresh daily limit and weekly limit 
+        //check daily limit and weekly limit
+        //If either are reached reject transaction
+        if (!(await checkDailyLimit(addressDetails))){
+            return false;
+        }
+        if (!(await checkWeeklyLimit(addressDetails))){
+            return false;
+        }
+        //refresh norequests
+        const norequests = await resetNoRequests(addressDetails);
+        if (norequests === 0){
+            await updateCounts(addressDetails, topUpAmount);
+            return true
+        } 
+        addressDetails = (await checkAddressExists(address))[0];
```

Comment:
Why are we sure that there is at least 1 entry in these rows?

Reply:
An address which exists will always have a atleast one entry if it does not exist the code will never reach this point. However if there is any error the new try catch block should handle it!